mlflow/types/utils.py

```diff
@@ -238,30 +238,6 @@ def _infer_schema(data: Any) -> Schema:
     """
     from scipy.sparse import csc_matrix, csr_matrix
 
-    # List[Dict[str, Union[DataType, List, Dict]]]
-    # e.g.
-    # [{'output': 'some sentence', 'ids': ['id1', 'id2'], 'dict': {'key': 'value'}},
-    #  {'output': 'some sentence', 'ids': ['id1', 'id2'], 'dict':
-    # {'key': 'value', 'key2': 'value2'}}]
-    # The corresponding pandas DataFrame representation should be `pd.DataFrame(data)`
-    #           output         ids                                dict
-    # 0  some sentence  [id1, id2]                    {'key': 'value'}
-    # 1  some sentence  [id1, id2]  {'key': 'value', 'key2': 'value2'}
-    # inferred schema -->
-    # Schema([ColSpec(type=DataType.string, name='output'),
-    #        ColSpec(type=Array(dtype=DataType.string), name='ids'),
-    #        ColSpec(type=Object([Property(name='key', dtype=DataType.string),
-    #                             Property(name='key2', dtype=DataType.string, required=False)]
-    #               ), name='dict')])
-    if isinstance(data, dict) and any(not isinstance(value, str) for value in data.values()):
-        # TODO: Add a link to docs/examples
-        _logger.info(
```

Comment:
note: Unrelated to this PR, but we found this message is no-longer required, during the same discussion with PM.

Reply:
Can we isolate this in a separate PR? We don't want to revert this when we need to revert this PR.

Reply:
Sure, will file a new PR for it

Reply:
Filed https://github.com/mlflow/mlflow/pull/12212